### PR TITLE
Nf 334 audit amend

### PR DESCRIPTION
**Pull request by rorticus0502**

### CODE REVIEW

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala [hunk 1]
```diff
@@ -71,6 +77,10 @@ class ClaimService @Inject() (auditConnector: AuditConnector, connector: NIDACCo
     )
 
     connector.amendClaim(AmendEISClaimRequest(eisRequest, amendClaim.uploads), correlationId)
+      .map { response =>
+        auditConnector.sendExplicitAudit("AmendClaim", AmendClaimAudit(response.error.isEmpty, amendClaim, response))
```
Comment (crosslandwa) [inline app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala:81]: CLARIFICATION: Will this code be executed in the result of a 500 response from the backend?

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimServiceSpec.scala [hunk 2]
```diff
@@ -46,6 +47,7 @@ class ClaimServiceSpec extends UnitSpec with BeforeAndAfterEach with TestData {
     super.beforeEach()
 
     when(nidacConnector.submitClaim(any(), any())(any())).thenReturn(Future.successful(validCreateClaimResponse))
+    when(nidacConnector.amendClaim(any(), any())(any())).thenReturn(Future.successful(validAmendClaimResponse))
```
Comment (njmoorewitchcat) [inline test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimServiceSpec.scala:50]: OBSERVATION: In regards to the clarification comment above, should there be a tests that do a Future.failure() return here and then that would explain what the audit would do under these circumstances - i.e. if it should still audit if the upstream fails or not.

File: app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala (hunk 1 above)
Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala:81]: No

Comment (rorticus0502) [reply app/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimService.scala:81]: In line with what we discussed with Nathan. If it is rejected we will audit it, but if some unexpected failure downstream, we wont, and that will be covered by implicit audits.

File: test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimServiceSpec.scala (hunk 2 above)
Comment (rorticus0502) [reply test/uk/gov/hmrc/nationalimportdutyadjustmentcentrefrontend/services/ClaimServiceSpec.scala:51]: Test for failed future added to document expected behaviour

**Review by crosslandwa — APPROVED** [review]

**Pull request merged**